src/Reporter.ts

```diff
@@ -133,7 +133,13 @@ class Reporter {
             }
         });
 
-        await Promise.all(reportPromises).catch((err) => {
+        await Promise.all(reportPromises).catch(async (err) => {
+            if (err.code === 'ECONNRESET') {
+                ThundraLogger.getInstance().debug(
+                    'Keep Alive connection reset by server. Will send monitoring data again.');
+                await this.sendReports();
```

Comment:
It is better to try only once. Not sure whether it is possible in practice, but in theory, this might result infinite recursion by calling `sendReports` recursively. To prevent this:

```javascript
async sendReports(dontRetry: boolean): Promise<void> {
    ...
    await Promise.all(reportPromises).catch(async (err) => {
            if (dontRetry != true) {
                if (err.code === 'ECONNRESET') {
                    ThundraLogger.getInstance().debug('Connection reset by server. Will send monitoring data again.');
                    await this.sendReports(true);
                    return;
                }
            }
            ThundraLogger.getInstance().error(err);
        });
}
```

Reply:
I included `connectionRetryCount` to Reporter to make sure we try to reconnect to server only once.